src/server/game/AntiCheat/AntiCheatMgr.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+Nemesis Anti Cheat
+Inspired by the old passive anti cheat (https://github.com/TrinityCore/TrinityCoreCustomChanges/tree/3.3.5-passive_anticheat)
+Authors:
+https://github.com/sveN295
+https://github.com/Jinnaix
+With feedback and support from https://github.com/acidmanifesto
+*/
+
+#ifndef SC_ACMGR_H
+#define SC_ACMGR_H
+
+#include "Common.h"
+#include "SharedDefines.h"
+#include "ScriptMgr.h"
+#include "Chat.h"
+#include "Player.h"
+#include "World.h"
+
+class Player;
+
+class TC_GAME_API AntiCheatMgr
+{
+public:
+    AntiCheatMgr(Player* player);
+    ~AntiCheatMgr();
+
+    void HackDetection(MovementInfo movementInfo, uint32 opcode);
+    void SetAllowedMovement(bool allowed);
+
+    Player* player;
```

Comment:
It might be better to rename this to `_player` for consistency with the other fields